### PR TITLE
fix: do not crash if "_" is parsed in builtins

### LINKED ISSUE #9802: Can't use _ in typeshed
As has been recommended as a workaround, I've tried using a custom typeshed directory to modify the builtins that Mypy sees, so that it recognises the `_()` function.

When adding to builtins.pyi:
```
from gettext import gettext
_ = gettext
```

I get:
```
error: INTERNAL ERROR -- Please try using mypy master on Github:
https://mypy.rtfd.io/en/latest/common_issues.html#using-a-development-mypy-build
Please report a bug at https://github.com/python/mypy/issues
version: 0.790
Traceback (most recent call last):
  File "mypy/semanal.py", line 4700, in accept
  File "mypy/nodes.py", line 1141, in accept
  File "mypy/semanal.py", line 3186, in visit_return_stmt
  File "mypy/nodes.py", line 1545, in accept
  File "mypy/semanal.py", line 3537, in visit_call_expr
  File "mypy/nodes.py", line 1545, in accept
  File "mypy/semanal.py", line 3454, in visit_call_expr
  File "mypy/nodes.py", line 1477, in accept
  File "mypy/semanal.py", line 3382, in visit_name_expr
  File "mypy/semanal.py", line 3895, in lookup
IndexError: string index out of range
```

Also, when I try:
```
from gettext import gettext as _
```

I don't get a crash, but Mypy still doesn't recognise it as a global function. Still producing dozens of:
`error: Name '_' is not defined  [name-defined]`

### PR DESCRIPTION
**Pull request by StefanM-TT**
allows the definiton of "_" as function in builtins (e.g via typeshed).

fixes issue #9802

### Description

Fixes #9802

### CODE REVIEW

File: mypy/semanal.py [hunk 1]
```diff
@@ -4296,7 +4296,7 @@ def lookup(self, name: str, ctx: Context,
             assert isinstance(b.node, MypyFile)
             table = b.node.names
             if name in table:
-                if name[0] == "_" and name[1] != "_":
+                if len(name) > 1 and name[0] == "_" and name[1] != "_":
```
Comment (sobolevn) [inline mypy/semanal.py:4299]: Question: why would we need to define `_` as a function in `__builtins__`? There's no such function as far as I remember.

The original issue was about a custom module, not `__builtins__`.

**Review by sobolevn — COMMENTED** [review]
Can you please also add a test case to make sure that this feature works as intended? 🙂

File: mypy/semanal.py (hunk 1 above)
Comment (JelleZijlstra) [reply mypy/semanal.py:4299]: It's a standard pattern: https://docs.python.org/3/library/gettext.html#gettext.install

**Comment by hauntsaninja** [comment]
Thank you!

(I was happy to merge without a dedicated test, not really worth a custom builtins stub)

**Pull request merged**